tensorboard/plugins/hparams/backend_context.py

```diff
@@ -326,16 +326,25 @@ def _compute_hparam_info_from_values(self, name, values):
             return None
 
         if result.type == api_pb2.DATA_TYPE_STRING:
-            distinct_values = set(
+            distinct_string_values = set(
                 _protobuf_value_to_string(v)
                 for v in values
                 if _can_be_converted_to_string(v)
             )
-            result.domain_discrete.extend(distinct_values)
+            result.domain_discrete.extend(distinct_string_values)
 
         if result.type == api_pb2.DATA_TYPE_BOOL:
             result.domain_discrete.extend([True, False])
 
+        if result.type == api_pb2.DATA_TYPE_FLOAT64:
+            # Always uses interval domain type for numeric hparam values.
+            distinct_float_values = sorted(
+                [v.number_value for v in values if v.HasField("number_value")]
```

Comment:
Is the `if v.HasField("number_value")` necessary? Seems like the logic L312 to L322 assures that if result.type == api_pb2.DATA_TYPE_FLOAT64 then all the values have a number_value field.

Reply:
Great point! Removed the field check.